### PR TITLE
Date toString tests

### PR DESCRIPTION
**Pull request by littledan**
These tests check the semantics of Date.prototype.toString, Date.prototype.toTimeString,
Date.prototype.toDateString and Date.prototype.toUTCString. There are probably ways
to make the more specific, but this is a start. The semantics are currently implementation-defined,
but a spec pull request at https://github.com/tc39/ecma262/pull/848 gives them this behavior.

### CODE REVIEW

File: test/built-ins/Date/prototype/toDateString/format.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  5. Return DateString(_t_).
+---*/
+
+let dateRegExp = /^(Sun|Mon|Tue|Wed|Thu|Fri|Sat) (Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec) [0-9]{2} [0-9]{4}$/
+let match = dateRegExp.exec(new Date().toDateString());
+assert.notSameValue(null, match);
+
+// Years are padded to the left with zeroes
+match = stringRegExp.exec(new Date('0020-01-01T00:00:00Z').toDateString());
```
Comment (leobalter) [inline test/built-ins/Date/prototype/toDateString/format.js]: s/stringRegExp/dateRegExp

**Comment by littledan** [comment]
The spec change landed, but changes still needed on the test.

File: test/built-ins/Date/prototype/toDateString/format.js (hunk 1 above)
Comment (littledan) [inline test/built-ins/Date/prototype/toDateString/format.js]: Fixed

**Comment by littledan** [comment]
With that typo fixed, the tests pass on a hacked version of V8 that implements the new specification, so I don't think there are any  more obvious typos.

**Pull request merged**